api/nim/v1alpha1/account_types.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2024 Red Hat, Inc.
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type (
+	// AccountSpec defines the desired state of an Account object.
+	AccountSpec struct {
+		// A reference to the Secret containing the NGC API Key.
+		SecretRef corev1.ObjectReference `json:"secretRef"`
+	}
+
+	// AccountStatus defines the observed state of an Account object.
+	AccountStatus struct {
+		// A reference to the Template for NIM ServingRuntime.
+		TemplateRef *corev1.ObjectReference `json:"templateRef,omitempty"`
+		// A reference to the ConfigMap with data for NIM deployment.
+		ConfigMapRef *corev1.ObjectReference `json:"configMapRef,omitempty"`
```

Comment:
nitpick: I would name this deploymentConfigMap or similar, but this is personal preference. Feel free to ignore.